### DISCUSSION TITLE
Setting a column with a value from another column json key value

### DISCUSSION THREAD

**Discussion by squidgemonster** [supabase/supabase#20411] [Questions] [answered]
I thought this simple query would work for this... moving a current json value to a standalone column for better performance - but running this doesn't seem to work:

```
update
  table_name
set
  column_name = jsonb_column ->> 'object_key_name'
where
  column_name = null;
```

Am I missing something? 🫤

**Comment by GaryAustin1** [comment, marked as answer]
You can't do = null.  You have to do IS null.  You can also use a Postgres generated column to set that column https://www.postgresql.org/docs/current/ddl-generated-columns.html

**Reply by squidgemonster** [reply]
Thanks @GaryAustin1 - I tried both = and IS - but neither seemed to update the table when I used the SQL UI on Superbase...

When I run the command it simply says 'Success. No rows returned'... and there's no change on the table. Unless I'm doing something wrong?

Great to know about generated columns too!

**Reply by squidgemonster** [reply]
Turns out that putting public. infront of my query was the culprit (and using IS instead of =)

Those are a few hours of my life I'm never getting back 😂
